pkg/reconciliation/reconcile_fql.go

```diff
@@ -1,98 +1,53 @@
 package reconciliation
 
 import (
-	"encoding/json"
 	"errors"
-	"fmt"
-	"strconv"
-	"strings"
 
 	"github.com/k8ssandra/cass-operator/pkg/httphelper"
 	"github.com/k8ssandra/cass-operator/pkg/internal/result"
 )
 
-// parseFQLFromConfig parses the DC config field to determine whether FQL should be enabled based on the presence of full_query_logging_options within the cassandra-yaml field.
-// To ease integration into the reconciliation process, it returns (shouldFQLBeEnabled, serverMajorVersion, ReconcileResult) where the ReconcileResult may be result.Error() or result.Continue().
-func parseFQLFromConfig(rc *ReconciliationContext) (bool, int64, result.ReconcileResult) {
+// CheckFullQueryLogging sets FQL enabled or disabled based on the `enableFQL` parameter, and takes serverMajorVersion for additional validation.
+// It calls the NodeMgmtClient which calls the Cassandra management API and returns a result.ReconcileResult.
+func (rc *ReconciliationContext) CheckFullQueryLogging() result.ReconcileResult {
 	dc := rc.GetDatacenter()
-	serverMajorVersion, err := strconv.ParseInt(strings.Split(dc.Spec.ServerVersion, ".")[0], 10, 8)
+	enableFQL, err := dc.FullQueryEnabled()
```

Comment:
I'm a bit confused in this part. It looks like we are determining whether to enable FQL, but we aren't validating whether we actually can validate FQL. 

I'm wondering what happens if the user is not using the webhook (which does do the validation) and tries to enable FQL on DSE (for example). We'd end up in a requeue loop - which is fine - but we should probably error out?

Should we consider calling `ValidateFQLConfig()` in the controllers as well as the webhooks to avoid that sort of issue?

Reply:
The ValidateFQLConfig() is called in the reconciliation method, it's done in the isValid.

Reply:
Ah, OK sorry. I missed that as I haven't played around in `handler` too much. You're right, ignore me :)